File: JAVA/level2/점프와_순간_이동.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        int ans = 0;
        int val;

        while(n != 0){
            val = 1;

            while(val*2 <= n)
                val *= 2;

            ans++;
            n -= val;
        }

        return ans;
    }
}

File: JavaScript/level2/점프와_순간_이동.js
function solution(n)
{
    var ans = 0;
    let val = 0;
    
    while(n != val){
        let temp = 1;
        ans++;
        while(n >= temp + val)
            temp *= 2;
        
        val += (temp/2);
    }
    
    return ans;
}